app/features/syllabus_generator/tools.py

```diff
@@ -0,0 +1,138 @@
+from typing import List, Tuple, Dict, Any
+
+import os
+import json
+import time
+
+from langchain_core.prompts import PromptTemplate
+from langchain_core.output_parsers import JsonOutputParser
+from langchain_core.pydantic_v1 import BaseModel, Field, ValidationError
+from langchain_google_genai import GoogleGenerativeAI
+from langchain_core.exceptions import OutputParserException
+
+
+from app.services.logger import setup_logger
+
+
+
+logger = setup_logger(__name__)
+
+def read_text_file(file_path):
+    # Get the directory containing the script file
+    script_dir = os.path.dirname(os.path.abspath(__file__))
+
+    # Combine the script directory with the relative file path
+    absolute_file_path = os.path.join(script_dir, file_path)
+    
+    with open(absolute_file_path, 'r') as file:
+        return file.read()
+
+class Syllabus:
+    def __init__(self, model, attributes, prompt_template, parser, verbose = False):
+        self.attributes = attributes
+        self.order =['course_title', 'grade_level', 'course_description', 'objectives_topics', 'required_materials', 'num_weeks', 'course_outline', 'grading_policy', 'class_policy', 'customization']
+        self.fields_dict = {
```

Comment:
This dict can be refactored with the Pydantic schema and the description inside of the Fields of each attribute.